File: app/src/main/java/io/github/zwieback/familyfinance/business/article/fragment/ExpenseArticleFragment.java
package io.github.zwieback.familyfinance.business.article.fragment;

import android.os.Bundle;

import io.github.zwieback.familyfinance.business.article.adapter.ExpenseArticleAdapter;
import io.github.zwieback.familyfinance.business.article.filter.ArticleFilter;

import static io.github.zwieback.familyfinance.business.article.filter.ExpenseArticleFilter.EXPENSE_ARTICLE_FILTER;

public class ExpenseArticleFragment extends ArticleFragment<ExpenseArticleAdapter> {

    public static ExpenseArticleFragment newInstance(ArticleFilter filter) {
        ExpenseArticleFragment fragment = new ExpenseArticleFragment();
        Bundle args = Companion.createArguments(EXPENSE_ARTICLE_FILTER, filter);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    protected ExpenseArticleAdapter createEntityAdapter() {
        ArticleFilter filter = extractFilter(EXPENSE_ARTICLE_FILTER);
        return new ExpenseArticleAdapter(requireContext(), getClickListener(), getData(), filter);
    }
}

File: app/src/main/java/io/github/zwieback/familyfinance/business/article/fragment/ExpenseArticleFragment.kt
package io.github.zwieback.familyfinance.business.article.fragment

import io.github.zwieback.familyfinance.business.article.adapter.ExpenseArticleAdapter
import io.github.zwieback.familyfinance.business.article.filter.ArticleFilter
import io.github.zwieback.familyfinance.business.article.filter.ExpenseArticleFilter.Companion.EXPENSE_ARTICLE_FILTER

class ExpenseArticleFragment : ArticleFragment<ExpenseArticleAdapter>() {

    override fun createEntityAdapter(): ExpenseArticleAdapter {
        val filter = extractFilter(EXPENSE_ARTICLE_FILTER)
        return ExpenseArticleAdapter(requireContext(), clickListener, data, filter)
    }

    companion object {
        fun newInstance(filter: ArticleFilter) = ExpenseArticleFragment().apply {
            arguments = createArguments(EXPENSE_ARTICLE_FILTER, filter)
        }
    }
}
